lightning/src/ln/outbound_payment.rs

```diff
@@ -721,8 +718,8 @@ impl OutboundPayments {
 			}
 		};
 		for path in route.paths.iter() {
-			if path.len() == 0 {
-				log_error!(logger, "length-0 path in route");
+			if path.hops.len() == 0 {
```

Comment:
In the case that we're the intro node, we'll want to advance the blinded path *before* pathfinding (otherwise how can we pathfind to ourselves?). So I think we-are-the-intro-node should be an error in payment parameters initialization. 

Therefore, I think `hops.len()` will always be `> 0`? If that's not the case, I have a bunch of `path.hops.first().unwrap()`s to update..